source/distributed/grid_tools.cc

```diff
@@ -0,0 +1,184 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+
+#include <deal.II/base/bounding_box.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/point.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/distributed/tria.h>
+#include <deal.II/distributed/grid_tools.h>
+
+#ifdef DEAL_II_WITH_P4EST
+
+#include <vector>
+#include <cmath>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace parallel
+{
+  namespace GridTools
+  {
+    template < int dim, int spacedim>
+    std::vector<BoundingBox<spacedim> >
+    compute_locally_owned_bounding_box
+    (const parallel::distributed::Triangulation< dim, spacedim > &distributed_tria)
```

Comment:
I don’t think there is anything specific to p::d::tria, u can accept p::tria

Reply:
Yes, this function (also the one I added "compute_cell_locally_owned_bounding_box") could work with any kind of triangulation but it would make no sense since, in that case, everything is locally owned or, if it is shared, we don't need a bounding box for a "general" description of the space since we already have it all.

Do you think it'd be better anyway to use a general MeshType?

Reply:
> would make no sense since, in that case, everything is locally owned

nope. First, you have parallel::shared::Triangulation, second, people can be using Triangulation but partitioning it manually. So what you really want to do here is: a) take standard Triangulation b) Add arbitrary predicate.

Reply:
> if it is shared, we don't need a bounding box for a "general" description of the space since we already have it all.

why is that? You don't know how tria is partitioned by Metis, it would still be useful to have a simple representation of the domain specified by a predicate (i.e. locally owned) even for p::s::Tria.